### ISSUE TITLE
bash hardcoded shell path not correct for FreeBSD systems

### ISSUE THREAD

**Issue by Cubox-** [smallstep/cli#146]
### Subject of the issue
Cannot compile/install on FreeBSD

### Your environment
* OS - FreeBSD
* Version - All impacted

### Steps to reproduce
gmake in repository

### Expected behaviour
Not tell me gmake: /bin/bash: Command not found

### Actual behaviour
gmake: /bin/bash: Command not found

### Additional context
On FreeBSD, bash is not installed by default. It can be installed through the ports system, but it will be located at /usr/local/bin/bash.
Does the project really need bash, or could it use sh instead?

**Comment by maraino** [comment]
@Cubox- step is now compilable in FreeBSD, but the install functionality won't work.

**Comment by Cubox-** [comment]
Great, thanks!
I tried to so something similar yesterday, but I mostly fought against dep not wanting my modifications to stick (not worked with dep before).

I'll let you know if I have any issue with step on FreeBSD (while keeping in mind install won't work).

**Comment by maraino** [comment]
👍 we want to move to `go mod` at some point

**Comment by Cubox-** [comment]
After updating the code, I was able to compile step using /bin/sh as shell.
Unless I missed something, you should be able to switch to /bin/sh without an issue (and avoid having to use a different shell per operating system)

I also had an issue using make install:
```
<cubox@Onyx>-<~/go/src/github.com/smallstep/cli> λ gmake clean build install
You will need to run 'make bootstrap' or 'dep ensure' directly to re-download any dependencies.
Build Complete!
usage: install [-bCcpSsUv] [-f flags] [-g group] [-m mode] [-o owner]
               [-M log] [-D dest] [-h hash] [-T tags]
               [-B suffix] [-l linkflags] [-N dbdir]
               file1 file2
       install [-bCcpSsUv] [-f flags] [-g group] [-m mode] [-o owner]
               [-M log] [-D dest] [-h hash] [-T tags]
               [-B suffix] [-l linkflags] [-N dbdir]
               file1 ... fileN directory
       install -dU [-vU] [-g group] [-m mode] [-N dbdir] [-o owner]
               [-M log] [-D dest] [-h hash] [-T tags]
               directory ...
gmake: *** [make/common.mk:128: install] Error 64
```

Adding to that, it should try to install on /usr/local on FreeBSD, but that's not a huge issue.

**Comment by maraino** [comment]
I'll let @dopey, he knows the build system better than me.

**Comment by Cubox-** [comment]
The same modification had to be done for step-ca. After that, all built fine as well.

**Comment by dch** [comment]
`perl -pi -e 's,bin/bash,usr/local/bin/bash,' $(find * -type f -name \*.go)` does the trick, for here & step-ca.

**Comment by dopey** [comment]
We'll try to get this fixed up and, at the very least, in master for both repos early this week. Thanks for the help in solving :)

**Comment by dopey** [comment]
Fixed by 71ee01bbfa698835c9bfac73458a95113181abef
